### DISCUSSION TITLE
Check Uplaod File Type in TypeScript

### DISCUSSION THREAD

**Discussion by MohMehrnia** [quasarframework/quasar#17604] [General - Components / Directives / etc]
i used this code check check file type:

but in typescript and tsc-vue show me error

```
const checkFileType = (files: []) => {
  return files.filter((file: { type: any; }) => file.type === 'video/mp4');
};

```

how do convert to typescript

**Comment by yusufkandemir** [comment]
I am guessing you are using the `filter` prop of `q-uploader`.

```ts
// :filter="checkFileType"
const checkFileType = (files: readonly File[] | FileList) => {
  return Array.from(files).filter((file) => file.type === 'video/mp4');
};
```

But, type typing of `QUploader['filter']` and possibly some other props/events shouldn't be `readonly any[] | FileList` but `readonly File[]` instead. So, it should be fixed. **After the fix**(not now), you will be able to this:
```ts
import { QUploaderProps } from 'quasar';

const checkFileType: QUploaderProps['filter'] = (files) => {
  return files.filter((file) => file.type === 'video/mp4');
};
```
